src/common/get-card-selection-view-data.ts

```diff
@@ -46,10 +46,9 @@ export const getCardSelectionViewData: GetCardSelectionViewData = (
     viewData.visualHelperEnabled = cardSelectionStoreData.visualHelperEnabled || false;
     viewData.expandedRuleIds = getRuleIdsOfExpandedRules(cardSelectionStoreData.rules);
 
-    const allResultUids = getAllResultUids(
-        unifiedScanResultStoreData.results.filter(resultsFilter),
-        cardSelectionStoreData.rules,
-    );
+    const filteredResults = unifiedScanResultStoreData.results.filter(resultsFilter);
+    const filteredResultUids = filteredResults.map(res => res.uid);
+    const allResultUids = getAllResultUids(filteredResultUids, cardSelectionStoreData.rules);
```

Comment:
I got confused by these names; my expectation would be that `filteredResultUids` is a subset of `allResultUids`, not the other way around.

Reply:
I'm not sure I'd expect there to be any reason for the rest of the implementation to actually use `filteredResults` instead of the version also filtered by the selectionStore rules, so maybe just combine these lines into something like `candidateResults` or `allCandidateResults`?

Reply:
Turns out I needed what was `filteredResults` to do your suggested change to `getResultsWithUnavailableHighlightStatus`, so I kept that. I like the candidate name idea instead of filtered, so I changed to that!